src/main/java/com/connexta/search/index/IndexManagerImpl.java

```diff
@@ -39,24 +41,25 @@ public void index(
     try {
       idAlreadyExists = crudRepository.existsById(productId);
     } catch (final Exception e) {
-      throw new IndexException("Unable to query index", e);
+      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to query index", e);
     }
     if (idAlreadyExists) {
-      throw new IndexException("Product already exists. Overriding is not supported");
+      throw new IndexException(
+          INTERNAL_SERVER_ERROR, "Product already exists. Overriding is not supported");
```

Comment:
✏️ I think this case should return 400. This can probably be handled with a new subclass of `IndexException`, maybe `OverridingIndexException`.